shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -77,52 +185,272 @@ - (BOOL)canBecomeFirstResponder {
   return YES;
 }
 
+#pragma mark - UITextInput Overrides
+
+- (id<UITextInputTokenizer>)tokenizer {
+  if (_tokenizer == nil) {
+    _tokenizer = [[UITextInputStringTokenizer alloc] initWithTextInput:self];
+  }
+  return _tokenizer;
+}
+
+- (UITextRange*)selectedTextRange {
+  return [[_selectedTextRange retain] autorelease];
+}
+
+- (void)setSelectedTextRange:(UITextRange*)selectedTextRange {
+  if (_selectedTextRange != selectedTextRange) {
+    UITextRange* oldSelectedRange = _selectedTextRange;
+    _selectedTextRange = [selectedTextRange copy];
+    [oldSelectedRange release];
+    [self updateEditingState];
+  }
+}
+
+- (NSString*)textInRange:(UITextRange*)range {
+  NSRange textRange = ((FlutterTextRange*)range).range;
+  return [self.text substringWithRange:textRange];
+}
+
+- (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
+  NSRange replaceRange = ((FlutterTextRange*)range).range;
+  NSRange selectedRange = ((FlutterTextRange*)self.selectedTextRange).range;
```

Comment:
Do you need to go through the accessor instead of just directly accessing the ivar? You wouldn't have to cast in that case. This interface is not meant to be safe for subclassing so there is nothing to be gained. Once you fix the copy in selectedTextRange, this will also yield an unnecessary copy.